server/app/services/program/ProgramService.java

```diff
@@ -44,6 +44,18 @@ public interface ProgramService {
   /** Get the data object about the programs that are in the active or draft version. */
   ActiveAndDraftPrograms getActiveAndDraftPrograms();
 
+  /**
+   * Update all {@link QuestionDefinition}s in a list of {@link ProgramDefinition}s asynchronously,
+   * by querying for questions then updating each {@link ProgramDefinition}s.
+   *
+   * @param programDefinitions the list of program definitions that should be updated
+   * @return a list of updated {@link ProgramDefinition}s with all of its associated questions if
+   *     they exist, or a QuestionNotFound is thrown when the future completes and a question is not
+   *     found.
```

Comment:
**suggestion (blocking):** use `@throws` instead of mentioning the error in the `@return` section

Reply:
Updated.

Reply:
Oh huh, yeah that's a bit of a surprise. Ah well, sorry for the noise